docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+import ConceptImage from '@site/src/components/conceptImage'
+
+# Message delivery guarantees
+
+TON is an asynchronous blockchain with complex structure which is very different from other blockchains. In this regard, new developers have questions about low-level things in TON. In this article we will have a look at one of these related to message delivery.
+
+## What is a message?
+
+If we look at Ethereum or almost any other synchronous blockchain, each transaction can contain several smart contract calls in it. For example, DEXs are performing multiple exchanges in one transaction if there are no liquidity for the selected trading pair.
```

Comment:
For example, DEXs are performing multiple exchanges in one transaction if there is no liquidity for the selected trading pair.